mqtt-examples/src/main/java/io/vertx/example/mqtt/app/Client.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright 2016 Red Hat Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.vertx.example.mqtt.app;
+
+import io.netty.handler.codec.mqtt.MqttQoS;
+import io.vertx.core.AbstractVerticle;
+import io.vertx.core.Vertx;
+import io.vertx.core.buffer.Buffer;
+import io.vertx.example.mqtt.util.Runner;
+import io.vertx.mqtt.MqttClient;
+import io.vertx.mqtt.MqttClientOptions;
+
+import java.nio.charset.Charset;
+
+/**
+ * An example of using the MQTT client
+ */
+public class Client extends AbstractVerticle {
+
+  private static final String MQTT_TOPIC = "/my_topic";
+  private static final String MQTT_MESSAGE = "Hello Vert.x MQTT Client";
+
+  // Convenience method so you can run it in your IDE
+  public static void main(String[] args) {
+    Runner.runExample(Client.class);
+  }
+
+  @Override
+  public void start() throws Exception {
+    MqttClientOptions options = new MqttClientOptions()
+      .setPort(1883)
+      .setHost("0.0.0.0")
+      .setKeepAliveTimeSeconds(2);
+
+    MqttClient client = MqttClient.create(Vertx.vertx(), options);
+
+
+    // handler will be called when we have a message in topic we subscribing for
+    client.publishHandler(publish -> {
+      System.out.println("Just received message on [" + publish.topicName() + "] payload [" + publish.payload().toString(Charset.defaultCharset()) + "] with QoS [" + publish.qosLevel() + "]");
+    });
+
+    // handle response on subscribe request
+    client.subscribeCompleteHandler(h -> {
+      System.out.println("Receive SUBACK from server");
+      System.out.print("Granted QoS levels: ");
+      h.grantedQoSLevels().forEach(System.out::print);
+      System.out.println();
```

Comment:
you don't need such foreach, grantedQoSLevels is a list of integer so just having : 

`System.out.println("Receive SUBACK from server with granted QoS : " + h.grantedQoSLevels())`

will print something like : 

Receive SUBACK from server with granted QoS : [0, 0, 1]